datafusion/core/tests/dataframe/mod.rs

```diff
@@ -1597,3 +1597,28 @@ async fn test_array_agg() -> Result<()> {
 
     Ok(())
 }
+
+#[tokio::test]
+async fn aggregate_with_alias() -> Result<()> {
+    let ctx = SessionContext::new();
+    let state = ctx.state();
+
+    let schema = Arc::new(Schema::new(vec![
+        Field::new("c1", DataType::Utf8, false),
+        Field::new("c2", DataType::UInt32, false),
+    ]));
+
+    let plan = scan_empty(None, schema.as_ref(), None)?
+        .aggregate(vec![col("c1")], vec![sum(col("c2"))])?
+        .project(vec![col("c1"), sum(col("c2")).alias("total_salary")])?
+        .build()?;
+
+    let plan = state.optimize(&plan)?;
+    let physical_plan = state.create_physical_plan(&Arc::new(plan)).await?;
+    assert_eq!("c1", physical_plan.schema().field(0).name().as_str());
+    assert_eq!(
+        "total_salary",
+        physical_plan.schema().field(1).name().as_str()
```

Comment:
Actually this test doesn't really use DataFrame API but creates logical plan and physical plan with `SessionState`. Maybe `physical-plan`? But there doesn't have `tests` module, so here is also good.

Reply:
You are right -- I mistakenly assumed that the `scan_empty` was returning a DataFrame (it is actually returning a `LogicalPlanBuilder`): https://github.com/alamb/arrow-datafusion/blob/9a15bf89682728a277cc0a8f443fbd4ba8398ae6/datafusion/core/src/test_util/mod.rs#L133

I will see if I can find a better home for this

Reply:
Moved in [2ea11a0](https://github.com/apache/arrow-datafusion/pull/8341/commits/2ea11a064b5ae7bb22b6a0e757382dee06e864e4)